gradle-jdks/src/main/java/com/palantir/gradle/jdks/DelayedConfigurableMap.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+import org.gradle.api.Action;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
+
+final class DelayedConfigurableMap<K, V> {
+    private final ProviderFactory providerFactory;
+    private final Supplier<V> valueFactory;
+    private final List<Provider<Map<K, Action<V>>>> configurations = new ArrayList<>();
```

Comment:
`>>>>` looks a lot like a merge conflict!

Reply:
This is how you tell it's good code